File: Java_ChildrenCompetitionSockets/CompetitionModel/src/main/java/competition/model/Entity.java
package competition.model;


import java.io.Serializable;
import java.util.Objects;

public class Entity<ID> implements HasId<ID>, Serializable {
    protected ID id;

    @Override
    public ID getId() {
        return id;
    }

    @Override
    public void setId(ID id) {
        this.id = id;
    }

    @Override
    public boolean equals(Object o){
        if(this == o){
            return true;
        }
        if(!(o instanceof Entity<?> entity)){
            return false;
        }
        return this.getId().equals(entity.getId());
    }

    @Override
    public int hashCode(){
        return Objects.hash(this.getId());
    }

    @Override
    public String toString() {
        return "Entity{" +
                "id=" + id +
                '}';
    }
}

File: CSharp_ChildrenCompetitionSockets/CSharp_ChildrenCompetitionSockets/model/Entity.cs
using System;

namespace CSharp_ChildrenCompetitionGUI.model
{
    [Serializable]
    public class Entity <ID> : HasId<ID>
    {
        public ID id { get; set; }

        public override string ToString()
        {
            return "Entity { " + "id=" + id + "}";
        }

        public override bool Equals(object obj)
        {
            if (this == obj)
            {
                return true;
            }

            if (!obj.GetType().IsInstanceOfType(typeof(Entity<ID>)))
            {
                return false;
            }
            Entity<ID> entity = (Entity<ID>) obj;
            return this.id.Equals(entity.id);
        }

        public override int GetHashCode()
        {
            return this.id.GetHashCode();
        }
    }
}